portals/publisher/src/main/webapp/source/src/app/components/GlobalPolicies/Policies/GlobalSpecificComponents/General.tsx

```diff
@@ -0,0 +1,458 @@
+
+    /**
+     * This function is triggered when the form is submitted for save.
+     * This handle the dynamic nature of the form, where the form fields are generated based on the policy spec.
+     * @param {React.FormEvent<HTMLFormElement>} event - Form submit event.
+     */
+    const submitForm = async (event: React.FormEvent<HTMLFormElement>) => {
+        event.preventDefault();
+        setSaving(true);
+        const updateCandidates: any = {};
+        Object.keys(state).forEach((key) => {
+            const value = state[key];
+            const attributeSpec = policySpec.policyAttributes.find(
+                (attribute: PolicySpecAttribute) => attribute.name === key,
+            );
+            if (value === null && getValueOfPolicyParam(key) && getValueOfPolicyParam(key) !== '') {
+                updateCandidates[key] = getValueOfPolicyParam(key);
+            } else if (value === null && attributeSpec?.defaultValue && attributeSpec?.defaultValue !==  null) {
+                updateCandidates[key] = attributeSpec.defaultValue;
+            } else {
+                updateCandidates[key] = value;
+            }
+        });
+
+        const globalPolicyToSave = {...globalPolicy};
+        globalPolicyToSave.parameters = updateCandidates;
+
+        updateGlobalOperations(globalPolicyToSave, currentFlow);
+
+        if (setDroppedPolicy) setDroppedPolicy(null);
+        setSaving(false);
+        handleDrawerClose();
+    };
+
+    /**
+     * Function to get the error string, if there are any errors. Empty string to indicate the absence of errors.
+     * @param {PolicySpecAttribute} specInCheck - The policy attribute that needs to be checked for any errors.
+     * @returns {string} - String with the error message, where empty string indicates that there are no errors. 
+     */
+    const getError = (specInCheck: PolicySpecAttribute) => {
+        let error = '';
+        const value = state[specInCheck.name];
+        if (value !== null) {
+            if (specInCheck.required && value === '') {
+                error = intl.formatMessage({
+                    id: 'Apis.Details.Policies.AttachedPolicyForm.General.required.error',
+                    defaultMessage: 'Required field is empty',
+                });
+            } else if (
+                value !== '' &&
+                specInCheck.validationRegex &&
+                !(!specInCheck.validationRegex || specInCheck.validationRegex === '')
+            ) {
+                // To check if the regex is a valid regex
+                try {
+                    if (!new RegExp(specInCheck.validationRegex).test(value)) {
+                        error = intl.formatMessage({
+                            id: 'Apis.Details.Policies.AttachedPolicyForm.General.regex.error',
+                            defaultMessage: 'Please enter a valid input',
+                        });
+                    }
+                } catch {
+                    // console.error(error);
```

Comment:
```suggestion
                    
```

Reply:
Addressed the required changes by [fddf57f](https://github.com/wso2/apim-apps/pull/515/commits/fddf57fa7308a53094c808e9bf2d2b40ed71b644) and [0ad490c](https://github.com/wso2/apim-apps/pull/515/commits/0ad490c497b08f999babbe521feaf83d029a6698)